### ISSUE TITLE
how to test own dataset on Demo.py 

### ISSUE THREAD

**Issue by Daktha123** [muelea/shapy#26]
Hello, I want to try my own dataset to test regressor. Which process that I need to do first before I run on regressor? Do I need to run openpose and get json format first? And also image size is only 300 450 px ? 

I run my own image and got this error : ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 135 and the array at index 1 has size 131

My image is :1920 1080 px but I resize to 300 450 px to match with yours and still have that error.

**Comment by neoglez** [comment]
I have the same problem. It seems that it is not possible to input an in-the-wild image to the network before adding to the target a significant amount of information such as the image can be transformed before it goes into the network.

**Comment by muelea** [comment]
We use OpenPose keypoints to get a person's bounding box which we use to center and crop them. If you have images where the person is already centered, you can skip this step and provide the bounding box directly. To do this, in the [data loader](https://github.com/muelea/shapy/blob/d7c98f0861652d1bbde82a3a76a30a917f79d231/regressor/human_shape/data/datasets/openpose.py#L430C13-L430C63), replace kpts_to_bbox() with the bounding box information [xmin, ymin, xmax, ymax]. I still recommend running OpenPose and use the automated crops, because these bounding boxes will match the ones the model was trained with. You can follow the images and openpose structure in our [demo](https://github.com/muelea/shapy/tree/master/samples) and then run the [SHAPY demo](https://github.com/muelea/shapy/tree/master#shapy-demo).

**Comment by neoglez** [comment]
@muelea Thanks for the helpful information. I will attempt to make a PR with a notebook demonstrating the complete pipeline.

**Comment by TharHtetAungg** [comment]
@muelea Hi, can I ask about which part of your code change to T pose ?  Could you please show me?

**Comment by felixshing** [comment]
> @muelea Thanks for the helpful information. I will attempt to make a PR with a notebook demonstrating the complete pipeline.

@neoglez Hey I would like to ask have you successfully done this pipeline that tests own dataset?

**Comment by rylzxc** [comment]
@muelea I have ran openpose on my 300 x 450 image and put the json and the image as per the folder structure in google colab and running the regressor code as you mentioned but it cant generate any visualisation or mesh for the image and openpose json I added. Am I missing a step?

**Comment by tillen1001** [comment]
I want to know how to run openpose.py to get human bounding box
is `cd shapy/regressor`
Then run `python -m human_shape.data.datasets.openpose` ?

**Comment by neoglez** [comment]
@felixshing Ja, I did but I had to make a lot of changes. I'm still trying to double check because the results I obtained are really strange.

**Comment by rshtern** [comment]
Can you explain how to train from an image without openpose?
